### PR TITLE
Add LOOC and change OOC color

### PR DESCRIPTION
**Pull request by Morb0**
## About the PR 

Add new `LOOC` chat channel that work in range same as `Local` but for OOC info.
Also use `LightSkyBlue` color for `LOOC` and for `OOC` use darker `RoyalBlue`.

Sometimes it is really useful, I saw how players write to IC adding brackets around message to represent OOC, sometimes you need to ask player nearby and write to OOC will be stupid, but I also think how ghosts write to LOOC...

**Screenshots**

Screenshot_4

**Changelog**

:cl:
- add: Added LOOC and changed OOC color

### CODE REVIEW

**Comment by moonheart08** [comment]
Surprisingly, I am not opposed to this.
I need to check though, does this work like normal speech where admins always see everything.

File: Content.Client/Chat/Managers/ChatManager.cs [hunk 1]
```diff
@@ -227,6 +229,8 @@ private void UpdateChannelPermissions()
             // Necessary so that we always have a channel to fall back to.
             DebugTools.Assert((SelectableChannels & ChatSelectChannel.OOC) != 0, "OOC must always be available");
             DebugTools.Assert((FilterableChannels & ChatChannel.OOC) != 0, "OOC must always be available");
+            DebugTools.Assert((SelectableChannels & ChatSelectChannel.LOOC) != 0, "LOOC must always be available");
+            DebugTools.Assert((FilterableChannels & ChatChannel.LOOC) != 0, "LOOC must always be available");
```
Comment (metalgearsloth) [inline Content.Client/Chat/Managers/ChatManager.cs:233]: Does LOOC need to be a fallback?

File: Content.Server/Chat/Managers/ChatManager.cs [hunk 2]
```diff
@@ -240,6 +240,35 @@ public void EntityMe(EntityUid source, string action)
...
         }
 
+        public void EntityLOOC(EntityUid source, string message)
+        {
+            // Check if entity is a player
+            if (!_entManager.TryGetComponent(source, out ActorComponent? actor))
+            {
+                return;
+            }
+
```
Comment (metalgearsloth) [inline Content.Server/Chat/Managers/ChatManager.cs:250]: If OOC is disabled then LOOC should be disabled as well IMO.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Chat/Managers/ChatManager.cs [hunk 3]
```diff
@@ -240,6 +240,35 @@ public void EntityMe(EntityUid source, string action)
...
+            // Check if message exceeds the character limit
+            if (message.Length > MaxMessageLength)
+            {
+                DispatchServerMessage(actor.PlayerSession, Loc.GetString("chat-manager-max-message-length-exceeded-message", ("limit", MaxMessageLength)));
+                return;
+            }
```
Comment (metalgearsloth) [inline Content.Server/Chat/Managers/ChatManager.cs:256]: This is silly, just combine it with the actor check like EntitySay

Comment (Morb0) [reply Content.Server/Chat/Managers/ChatManager.cs:266]: Check is OOC enabled need PlayserSession that exist in actor and now we:
- get actor
- check OOC enabled
- check message length
so it seems there is no need to combine actor getting and message length check

Comment (metalgearsloth) [reply Content.Server/Chat/Managers/ChatManager.cs:266]: Whisper PR is going to fix this anyway so doesn't matter.

**Comment by Morb0** [comment]
Added a separate command to toggle LOOC

**Review by moonheart08 — APPROVED** [review]

File: Content.Server/Chat/Commands/SetLOOCCommand.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        var looc = cfg.GetCVar(CCVars.LoocEnabled);
+
+        if (args.Length == 0)
+        {
+            looc = !looc;
+        }
+
+        if (args.Length == 1 && !bool.TryParse(args[0], out looc))
```
Comment (metalgearsloth) [inline Content.Server/Chat/Commands/SetLOOCCommand.cs:34]:
```suggestion
        if (args.Length > 1 || !bool.TryParse(args[0], out looc))
```

**Pull request merged**